azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java

```diff
@@ -24,15 +31,43 @@
     protected boolean isEnabled = true;
 
     public AppInsightsProxy(final TelemetryConfiguration config) {
-        // InstrumentationKey will be read from ApplicationInsights.xml
-        client = new TelemetryClient();
+        client = new TelemetryClient(readConfigurationFromFile());
         if (config == null) {
             throw new NullPointerException();
         }
         configuration = config;
         defaultProperties = configuration.getTelemetryProperties();
     }
 
+    /**
+     * This is a workaround for telemetry issue. ApplicationInsight read configuration file by JAXB, and JAXB parse
+     * configuration by JAXBContext, but the context model differs in Java 8 and Java 11 during maven execution, so
+     * read the config file by maven plugin here.
+     */
+    private com.microsoft.applicationinsights.TelemetryConfiguration readConfigurationFromFile() {
+        final com.microsoft.applicationinsights.TelemetryConfiguration telemetryConfiguration =
+            new com.microsoft.applicationinsights.TelemetryConfiguration();
+        final Map<String, String> channelProperties = new HashMap<>();
+        channelProperties.put(TelemetryChannelBase.FLUSH_BUFFER_TIMEOUT_IN_SECONDS_NAME, "1");
+        final TelemetryChannel channel = new InProcessTelemetryChannel(channelProperties);
+
+        telemetryConfiguration.setChannel(channel);
+        telemetryConfiguration.setInstrumentationKey(readInstrumentationKeyFromConfiguration());
+        return telemetryConfiguration;
+    }
+
+    private String readInstrumentationKeyFromConfiguration() {
```

Comment:
Why not just use `.propertes` to store the key instead of reading from xml.

Reply:
yes, nice idea.